### PR TITLE
Do precision comparison to prevent the creation of keyframes with a time that already exists

### PR DESCRIPTION
**Pull request by ghost**
Attempts to address:  https://github.com/godotengine/godot/issues/24908

Submitting for review.  Unclear if it is the ideal approach, because I'm not sure about the details or intentions regarding decimal truncation for keyframe time values.

There may yet be more to address with keyframe time precision beyond this issue.

### CODE REVIEW

File: scene/resources/animation.cpp [hunk 1]
```diff
@@ -815,15 +815,17 @@ int Animation::_insert(float p_time, T &p_keys, const V &p_value) {
 
 	while (true) {
 
-		if (idx == 0 || p_keys[idx - 1].time < p_time) {
-			//condition for insertion.
-			p_keys.insert(idx, p_value);
-			return idx;
-		} else if (p_keys[idx - 1].time == p_time) {
+		// Condition for replacement.
+		if (idx > 0 && Math::abs(p_keys[idx - 1].time - p_time) < CMP_EPSILON) {
```
Comment (akien-mga) [inline scene/resources/animation.cpp]: You could use `Math::is_equal_approx()` now (merged recently in master) to simplify this.

Comment (avencherus) [reply scene/resources/animation.cpp]: Thanks, I'll take a look into it.

Comment (avencherus) [reply scene/resources/animation.cpp]: @akien-mga Should be updated to the recommended convention.  In this case it was an approximately zero comparison, but seems that's provided in Math as well.

Comment (aaronfranke) [reply scene/resources/animation.cpp]: Aren't you comparing `p_keys[idx - 1].time` with `p_time`?

Comment (akien-mga) [reply scene/resources/animation.cpp]: Yes, it should be `Math::is_equal_approx(p_keys[idx - 1].time, p_time)`, avoiding these absolute differences comparisons to 0 is the all point of the method :)

**Review by aaronfranke — APPROVED** [review]
Note that the logic is changed to require `idx > 0` for the insert block, I'm not sure why / if that's desired, but the rest looks good.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**